src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -642,6 +659,60 @@ bool AreaTrigger::HasSplines() const
     return bool(_spline);
 }
 
+void AreaTrigger::InitCircularMovement(AreaTriggerCircularMovementInfo const& cmi, uint32 timeToTarget)
+{
+    // should be sent in object create packets only
+    m_uint32Values[AREATRIGGER_TIME_TO_TARGET] = timeToTarget;
+
+    _areaTriggerCircularMovementInfo = cmi;
+
+    _areaTriggerCircularMovementInfo.TimeToTarget = GetTimeToTarget();
+    _areaTriggerCircularMovementInfo.ElapsedTimeForMovement = GetElapsedTimeForMovement();
+
+    if (IsInWorld())
+    {
+        // TODO: broadcast AreaTriggerReShape?
+    }
+}
+
+Position const* AreaTrigger::GetCircularMovementCenterPosition() const
+{
+    if (!_areaTriggerCircularMovementInfo.Center.is_initialized() && _areaTriggerCircularMovementInfo.TargetGUID.is_initialized())
```

Comment:
which one takes priority? pos or guid?

Reply:
Tbh, I didnt see a areatrigger which had both, so I don't think it matters

Reply:
it matters, if you want clean code

Reply:
I mean there isn't any areatrigger which does have both. So why does it matter? If the areatrigger does have a targetGuid he takes that, if he doesn't he must have a center

Reply:
so, why do you check if it does not have a center pos?

Reply:
You're correct.. if it doesn't matter, then the initialized check for center isn't needed at this position.

I think it was done because the setter for center and targetGuid doesn't reset the counter part. I will handle that later

Reply:
the caller should handle that

Reply:
Why is that ? Should the caller manually remove targetGuid? If he wants to set center position ?